src/datahike/connector.cljc

```diff
@@ -15,52 +16,155 @@
+                                       :parent p}))
+                          commit-id))))))
+
+(defn create-commit-id [db]
+  (let [{:keys [hash max-tx max-eid config]
+         {:keys [parents]} :meta} db]
+    (if (:crypto-hash? config)
+      (uuid [hash max-tx max-eid parents])
+      (uuid))))
+
+(defn update-and-flush-db
+  ([connection tx-data tx-meta update-fn]
+   (update-and-flush-db connection tx-data tx-meta update-fn
+                        #{(get-in @connection [:config :branch])}))
+  ([connection tx-data tx-meta update-fn parents]
+   (let [parents               (branch-heads-as-commits (:store @connection) parents)
+         {:keys [db-after]
+          {:keys [db/txInstant]}
+          :tx-meta
+          :as   tx-report}     @(update-fn connection tx-data tx-meta)
+         {:keys [eavt aevt avet temporal-eavt temporal-aevt temporal-avet
+                 schema rschema system-entities ident-ref-map ref-ident-map config
+                 max-tx max-eid op-count hash meta]}
+         db-after
+         cid (create-commit-id db-after)
+         meta                  (assoc meta
+                                      :datahike/parents parents
+                                      :datahike/updated-at txInstant
+                                      :datahike/commit-id cid)
+         store                 (:store @connection)
+         backend               (di/konserve-backend (:index config) store)
+         ;; flush for in-memory backends would only make sense if multiple processes access the db
+         ;; TODO: update can also be skipped for external stores when single process writes unless memory should be freed up
+         ;;       -> add options to config
+         backend?              (not= :mem (-> config :store :backend))
+         eavt-flushed          (cond-> eavt backend? (di/-flush backend))
+         aevt-flushed          (cond-> aevt backend? (di/-flush backend))
+         avet-flushed          (cond-> avet backend? (di/-flush backend))
+         keep-history?         (:keep-history? config)
+         temporal-eavt-flushed (when keep-history? (cond-> temporal-eavt backend? (di/-flush backend)))
+         temporal-aevt-flushed (when keep-history? (cond-> temporal-aevt backend? (di/-flush backend)))
+         temporal-avet-flushed (when keep-history? (cond-> temporal-avet backend? (di/-flush backend)))
+         db-to-store (merge
+                      {:schema          schema
+                       :rschema         rschema
+                       :system-entities system-entities
+                       :ident-ref-map   ident-ref-map
+                       :ref-ident-map   ref-ident-map
+                       :config          config
+                       :meta            meta
+                       :hash            hash
+                       :max-tx          max-tx
+                       :max-eid         max-eid
+                       :op-count        op-count
```

Comment:
Could be made more condensed by using `select-keys` on db-after and merge it with the changes. I am just saying because you are refactoring this anyway in this PR.

Reply:
This is possible, but it also requires to transform the result of `select-keys` into a map, which makes the code a bit convoluted. I think in this case the number of lines saved is not worth the tradeoff of easy readability.